python/plugins/processing/gui/AlgorithmExecutor.py

```diff
@@ -329,8 +332,15 @@ def executeIterating(alg, parameters, paramToIter, context, feedback):
             if out.name() not in outputs:
                 continue
 
-            o = outputs[out.name()]
-            parameters[out.name()] = QgsProcessingUtils.generateIteratingDestination(o, i, context)
+            o = outputs[out.name()]            
+            if isinstance(o, QgsProperty):
```

Comment:
Could you use 

    QgsProcessingParameters::valueAsString(...)

Here instead of the duplicate code for checking input type?

Reply:
oh, good catch. Will change that :-)

Reply:
mmm, i dont't seem to get a value if i use that:

```dest = QgsProcessingParameters.parameterAsString(alg.parameterDefinition(out.name()), parameters, context)```

That gives me a null string all the time. 

Also, looking at the code of that method, it looks like it will never return TEMPORARY_OUTPUT, since it resolves it to a temp path in that case (see https://qgis.org/api/qgsprocessingparameters_8cpp_source.html#l00071)

Any other idea for replacing the type check?

Reply:
@nyalldawson any suggestion here, or can we merge with the current code?